src/provider.rs

```diff
@@ -117,6 +130,28 @@ impl CustomGetHandler for () {
     }
 }
 
+pub trait AuthorizationHandler: Send + Sync + Clone + 'static {
+    /// Handle the authorization request, given an opaque data blob from the requester.
+    fn authorize(
+        &self,
+        db: Database,
+        token: Option<AuthToken>,
+        request: Request,
+    ) -> BoxFuture<'static, anyhow::Result<()>>;
+}
+
+/// Define AuthorizationHandler for () so we can use it as a no-op default.
+impl AuthorizationHandler for () {
+    fn authorize(
+        &self,
+        _db: Database,
+        _token: Option<AuthToken>,
+        _request: Request,
+    ) -> BoxFuture<'static, anyhow::Result<()>> {
+        async move { Ok(()) }.boxed()
```

Comment:
I think the default handler should check that the auth token is actually `None`.